test/phoenix/router/helpers_test.exs

```diff
@@ -136,6 +137,35 @@ defmodule Phoenix.Router.HelpersTest do
     end
   end
 
+  test "url helper shows an error if an id is accidentally passed" do
+    error_message = """
+    Elixir.Phoenix.Router.HelpersTest.Router.Helpers.bottom_path/5 called with invalid params. The last argument to this function should be a keyword list or a map.
+    For example:
+
+      Router.bottom_path(conn, :bottom, :arg1, :arg2, page: 5, per_page: 10)
+
+    It is possible you have called this function without defining the proper number of path segments in your router.
+    """
```

Comment:
Let's use a regex and match only on the printed helper:

```
error_suggestion = ~r/Router.bottom_path\(conn, :bottom, :arg1, :arg2, page: 5, per_page: 10\)/
```

This way if we change minor details in the message we don't break the test.